### PR TITLE
fix(cheatcode): modify transfer object on prank if necessary

### PR DESCRIPTION
**Pull request by joshieDo**
ref  https://github.com/gakonst/foundry/issues/280

added test as well

### CODE REVIEW

**Comment by wilsoncusack** [comment]
🔥

File: evm-adapters/testdata/CheatCodes.sol [hunk 1]
```diff
@@ -186,6 +186,21 @@ contract CheatCodes is DSTest {
...
     }
 
+    function testPayable() public {
+        Prank prank = new Prank();
+        address new_sender = address(1337);
+        hevm.deal(new_sender, 10 ether);
+        
+        hevm.prank(new_sender);
+        prank.payableBar{value: 1 ether}(new_sender);
+        assertEq(new_sender.balance, 9 ether);
```
Comment (wilsoncusack) [inline evm-adapters/testdata/CheatCodes.sol:196]: want to check `address(this)` balance as well?

Comment (joshieDo) [reply evm-adapters/testdata/CheatCodes.sol:198]: for sure

File: evm-adapters/testdata/CheatCodes.sol [hunk 2]
```diff
@@ -186,6 +186,25 @@ contract CheatCodes is DSTest {
         prank.bar(address(this));
     }
 
+    function testPayable() public {
```
Comment (brockelmore) [inline evm-adapters/testdata/CheatCodes.sol:189]: can we change to `testPrankPayable` just so you can run  `-m testPrank` and test all prank related cheatcodes?

**Review by brockelmore — CHANGES_REQUESTED** [review]

File: evm-adapters/testdata/CheatCodes.sol (hunk 2 above)
Comment (joshieDo) [reply evm-adapters/testdata/CheatCodes.sol]: done

**Review by brockelmore — APPROVED** [review]
lgtm

**Pull request merged**